Formula/cloudprober.rb

```diff
@@ -0,0 +1,18 @@
+class Cloudprober < Formula
+  desc "Active monitoring software to detect failures before your customers do"
+  homepage "https://cloudprober.org"
+  url "https://github.com/cloudprober/cloudprober/archive/refs/tags/v0.12.0.tar.gz"
+  sha256 "e2289eaad263e35e20ed3afaccf4c854df179de4e36aa705fcf106d93a352b0b"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "make", "cloudprober", "VERSION=v#{version}"
+    bin.install "cloudprober"
+  end
+
+  test do
+    assert_match "v#{version}", shell_output("#{bin}/cloudprober --version").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Done.